sample_files/pre-commit-13.0/.pre-commit-config.yaml

```diff
@@ -18,20 +18,26 @@ repos:
     hooks:
       - id: black
   - repo: https://github.com/prettier/prettier
-    rev: "1.19.1"
+    rev: "2.0.2"
     hooks:
       - id: prettier
-  # TODO Avoid awebdeveloper/pre-commit-prettier if possible
-  # HACK https://github.com/prettier/prettier/issues/7407
-  - repo: https://github.com/awebdeveloper/pre-commit-prettier
-    rev: v0.0.1
-    hooks:
-      - id: prettier
-        name: prettier xml plugin
         additional_dependencies:
-          - "prettier@1.19.1"
           - "@prettier/plugin-xml@0.7.2"
-        files: \.xml$
+        args:
+          - --plugin=@prettier/plugin-xml
+        files: "\\.(\
+          css|less|scss\
+          |graphql|gql\
+          |html\
+          |js|jsx\
+          |json\
+          |md|markdown|mdown|mkdn\
+          |mdx\
+          |ts|tsx\
+          |vue\
+          |yaml|yml\
+          |xml\
+          )$"
```

Comment:
This is the only part I'd like to avoid. We can do it like this:

```yaml
   - repo: https://github.com/prettier/prettier

    rev: "2.0.2"
    hooks:
      - &prettier
        id: prettier
         additional_dependencies:
           - "@prettier/plugin-xml@0.7.2"
        args:
          - --plugin=@prettier/plugin-xml
      - <<: *prettier
        files: \\.xml$
```

Basically we'd be launching the same hook twice: once for the standard files supported by prettier, and once just for XML files.

Reply:
The problem with the prettier run without the xml plugin is that it warns about an unknown option: `[warn] Ignored unknown option { xmlWhitespaceSensitivity: "ignore" }. Did you mean htmlWhitespaceSensitivity?`.

Also I find it good to be explicit about the file types we put in prettier's hands.

Reply:
> The problem with the prettier run without the xml plugin is that it warns about an unknown option: `[warn] Ignored unknown option { xmlWhitespaceSensitivity: "ignore" }. Did you mean htmlWhitespaceSensitivity?`.

It wouldn't warn because the plugin would be loaded in both steps. Just no XML files would be selected in the 1st one.

> Also I find it good to be explicit about the file types we put in prettier's hands.

We don't do it for any other formatter, unless there's a problem with Prettier's selection that can't be fixed with exclude, I prefer to keep this shorter and more maintainable, just like it is.